refactor: Completely remove Blockstore

kdmccormick opened the pull request:
## Description / Supporting Info

Implements https://github.com/openedx/public-engineering/issues/238

Note: This also removes the `copy_library_from_v1_to_v2` management command, which has been broken ever since we switched from Blockstore to Learning Core.

## Testing

Mostly, I'm counting on linting and unit tests to confirm that this is a clean removal.

Dave, can you confirm with this branch that you can still create and use Learning-Core-Backed Content Libraries locally (to the same extent that you were able to before)?

## Merge Deadline

Before the Redwood cut, if possible.

Not a huge deal if it misses the cut, but it might muddy the waters to still have Blockstore references floating around in Redwood.

kdmccormick committed:
refactor: remove remaining blockstore references

kdmccormick committed:
refactor: delete blockstore_api

kdmccormick committed:
build: remove blockstore from requirement input files

ormsbee commented on openedx/core/djangoapps/content_libraries/views.py:
```diff
@@ -708,9 +710,12 @@ def put(self, request, usage_key_str, file_path):
+            #         In the future, we need a way to use file_wrapper.chunks() to read
+            #         the file in chunks and stream that to Blockstore, but Blockstore
+            #         currently lacks an API for streaming file uploads.
+            #       Ref:  https://github.com/openedx/edx-platform/issues/34737
```
We'll still need some kind of base checking to make sure it's not over our size limit, and we'll probably want to support streaming at some point.

github-actions[bot] committed:
build: recompile python dependencies to remove openedx-blockstore

Commit generated by workflow `kdmccormick/edx-platform/.github/workflows/compile-python-requirements.yml@refs/heads/master`

Co-authored-by: Kyle McCormick <kyle@axim.org>
